iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -1085,20 +1241,25 @@ protected override void Dispose(bool disposing)
 
                 if (!_isDisposed)
                 {
+                    _isDisposing = true;
+
                     base.Dispose(disposing);
                     if (disposing)
                     {
                         _handleDisconnectCts?.Cancel();
                         _handleDisconnectCts?.Dispose();
```

Comment:
If the CTS is disposed, will the cancellation token still be valid (used by the async tasks that are watching it, or might they throw when the token from it is accessed)?

Reply:
It works fine.

```csharp
namespace CtsDisposed
{
    public static class Program
    {
        public static async Task Main()
        {
            var cts = new CancellationTokenSource();
            Task work = DoWorkAsync(cts.Token);
            await Task.Delay(5000).ConfigureAwait(false);
            cts.Cancel();
            try
            {
                await work.ConfigureAwait(false);
            }
            catch (Exception ex)
            {
                await Console.Out.WriteLineAsync($"Caught {ex}");
            }
        }

        private static async Task DoWorkAsync(CancellationToken ct)
        {
            try
            {
                await Task.Delay(-1, ct).ConfigureAwait(false);
            }
            catch (OperationCanceledException)
            {
                await Console.Out.WriteLineAsync("DoWork cancelled");
            }
        }
    }
}
```

Reply:
The code comments in the CancellationTokenSource [sourcecode](https://github.com/dotnet/runtime/blob/0a15c3b97dabaf94e5d4399aef2fd48c2b3f446a/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs#L505-L510) supports your test:

```
// And, we tolerate Dispose being used concurrently with Cancel.  This is necessary
// to properly support, e.g., LinkedCancellationTokenSource, where, due to common usage patterns,
// it's possible for this pairing to occur with valid usage (e.g. a component accepts
// an external CancellationToken and uses CreateLinkedTokenSource to combine it with an
// internal source of cancellation, then Disposes of that linked source, which could
// happen at the same time the external entity is requesting cancellation).
```